src/validators/bid.validator.ts

```diff
@@ -1,30 +1,54 @@
-import { Game, PlayersBid } from '../game.interfaces';
+import { Game, PlayersBid, Phase } from '../game.interfaces';
 import { Bid } from '../game.actions';
 import { getNextTurn } from '../helpers/players.helpers';
 import { hasMarriage } from '../helpers/cards.helpers';
 import * as _ from 'lodash';
-import { isMaxBid, hasTwoPasses } from '../helpers/bid.helpers';
+import { isMaxBid, hasTwoPasses, getHighestBid, isAchievableBid, isValidBidValue } from '../helpers/bid.helpers';
 
 export function canBid(state: Game, action: Bid): boolean {
-    const playerId = action.player;
-    const playerCards = state.cards[playerId];
+    if (state.phase != Phase.BIDDING_IN_PROGRESS) {
+        return false;
+    }
+    if (isAchievableBid(action) == false) {
+        return false;
+    }
+    if (isValidBidValue(action) == false) {
```

Comment:
imho we should pass here bid instead of action. I know interfaces are equal, but it's tricky now. passing Bid interface, but function requires PlayerBid

Reply:
For me, following functions:
* `isMaxBid`
* `isValidBidValue`
* `isAchievableBid`

should have different signatures
not
`functionName({ bid }: PlayersBid)`
but
`functionName( bid : number)`
What do you think?